frontend/src/components/object/ObjectVersion.vue

```diff
@@ -0,0 +1,174 @@
+<script setup lang="ts">
+import { storeToRefs } from 'pinia';
+import { onMounted, ref, watch } from 'vue';
+
+import {
+  DeleteObjectButton,
+  DownloadObjectButton
+} from '@/components/object';
+import { Button, Column, DataTable, useToast } from '@/lib/primevue';
+import { useAppStore, useAuthStore, usePermissionStore, useUserStore, useVersionStore } from '@/store';
+import { Permissions, RouteNames } from '@/utils/constants';
+import { ButtonMode } from '@/utils/enums';
+import { formatDateLong } from '@/utils/formatters';
+
+import type { Ref } from 'vue';
+import type { User, Version } from '@/types';
+
+type VersionDataSource = {
+  createdByName?: string;
+} & Version;
+
+// Props
+type Props = {
+  bucketId: string;
+  objectId: string;
+  versionId: string;
+};
+
+const props = withDefaults(defineProps<Props>(), {});
+
+// Store
+const permissionStore = usePermissionStore();
+const userStore = useUserStore();
+const versionStore = useVersionStore();
+const { getUserId } = storeToRefs(useAuthStore());
+const { getUserSearch } = storeToRefs(userStore);
+
+// State
+const tableData: Ref<Array<VersionDataSource>> = ref([]);
+
+// Actions
+const toast = useToast();
+
+// eslint-disable-next-line no-unused-vars, @typescript-eslint/no-unused-vars
+const showInfo = async (id: string) => {
+  // emit('show-object-info', id);
+};
+
+function onDeletedSuccess() {
+  toast.add({
+    severity: 'success',
+    summary: 'Success',
+    detail: 'File deleted',
+    life: 3000
+  });
+}
+
+async function load() {
+  await versionStore.fetchVersions({objectId: props.objectId});
+  const versions = versionStore.findVersionsByObjectId(props.objectId);
+  await userStore.fetchUsers({userId:[versions.map( (x: Version) => x.createdBy)]});
+
+  tableData.value = versions.map( (v: Version) => {
+    return {
+      ...v,
+      createdByName: getUserSearch.value.find( (u: User) => u.userId === v.createdBy )?.fullName
+    };
+  });
+}
```

Comment:
Using a closure you can avoid needing to write the return:
```ts
  tableData.value = versions.map( (v: Version) => ({
    ...v,
    createdByName: getUserSearch.value.find( (u: User) => u.userId === v.createdBy )?.fullName
  }));
```
Generally recommend favoring this form of syntax over an explicit return unless there needs to be operations that have to be done prior to returning out of the mapping.